File: lessons/compiled_lang/cpp/src/b20.formatting.cpp
#include <cstdio>
#include <string>

int main() {
    int number = 5;
    char character = 'a';
    std::string text = "This is a string";

    std::printf("Number is %d.\nCharacter is '%c'.\nString is \"%s\".\n",
        number, character, text.c_str());

    return 0;
}

File: lessons/compiled_lang/java/src/b20.formatting.java
// Not public - see a00.output.java for why.
class B20Formatting {
    public static void main(String[] args) {
        int number = 5;
        char character = 'a';
        String text = "This is a string";

        // Positional format specifiers (%1$d, %2$c, %3$s) in one combined
        //  format string - distinct from b10's three separate printf calls
        //  (mirrors lessons/compiled_lang/rust's b20.formatting.rs, which makes the
        //  same "one call, positional {0}/{1}/{2}" choice for this lesson).
        //  Explicit "\n", not "%n" - %n renders as "\r\n" on a Windows JVM,
        //  which would break the exact-string comparison test.
        String output = String.format("Number is %1$d.\nCharacter is '%2$c'.\nString is \"%3$s\".\n",
                number, character, text);

        System.out.print(output);
    }
}
